teps/0081-convention-for-parameterresult-name-ownership-and-definition.md

```diff
@@ -0,0 +1,401 @@
+      should be applied to the workspace supplied via `kontext` (hypothetical).
+
+
+6. Producing an attestation
+
+    ```yaml
+      results:
+        - name: dev.tekton.chains.attestation.predicateType
+          description: The in-toto "predicateType" being emitted.
+        - name: dev.tekton.chains.attestation.predicate
+          description: The in-toto "predicate" being emitted.
+    ```
+
+    We can define a duck type with the elements we need to produce an attestation
+    (e.g. as a strawperson here, I'm using the in-toto style), where controllers
+    such as Chains can use this duck type to determine the attestation payload, and
+    other duck types such as `2.` to determine the artifact to which that attestation
+    applies (we could also have `dev.tekton.chains.subject`, but that's beyond the
+    scope of this TEP!).
+
+    The application of duck types over well defined naming conventions is much
+    better than the current ad hoc and fragile pattern matching in Chains, but to
+    improve it we really need a set of conventions that we use.
+
+    Perhaps once [the TEP for richer result types](https://github.com/tektoncd/community/pull/479)
+    lands this could be:
+
+    ```yaml
+    results:
+      - name: dev.tekton.chains.attestation
+        description: The in-toto statement being emitted.
+        schema:
+          type: object
+          properties:
+            predicateType: {
+              type: string
+            }
+            predicate: {
+              type: object
+            }
+    ```
+
+## Requirements
+
+Needs to be supported by Tekton Tasks/Pipelines
+(see [here](https://github.com/tektoncd/community/pull/503)).
+
+Needs to unambiguously identify the organization responsible,
+which should maintain a catalog documenting "well known" parameters,
+and their expected semantics (as appropriate).
+
+## Proposal
+
+This proposes the use of reverse-domain scoped names, e.g.
+`dev.tekton.foo.bar` would be owned by the Tekton mark
+holders and subdivided according to their will (e.g.
+`dev.tekton.pipelines.foo.bar` may be for `pipelines` use).
+
+### Notes/Caveats (optional)
+
+There are a number of precedents for the use of reverse domain-scoped names:
+
+1. Java imports (easily the most pervasive):
+
+    ```java
+    import com.google.common.collect.ImmutableMap;
+    ```
+
+    > Anecdotally, Google used this internally for Python package names as
+    > well, but this seems like a Google-specific convention for Python.
+
+2. CloudEvent [types](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#type):
+
+    ```
+    com.github.pull_request.opened
+    ```
+
+3. Containerd plugins (copied from [stargz-snapshotter](https://github.com/containerd/stargz-snapshotter#quick-start-with-kubernetes))
+
+    ```toml
+    # Use stargz snapshotter through CRI
+    [plugins."io.containerd.grpc.v1.cri".containerd]
+      snapshotter = "stargz"
+      disable_snapshot_annotations = false
+    ```
```

Comment:
Another example that just came up in slack is annotations on OCI images use reversed-domains: https://github.com/opencontainers/image-spec/blob/main/annotations.md#rules

Reply:
incorporated